contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,382 @@
+    }
+
+    function setMintingLicenseSpecForAll(
+        address ipId,
+        Licensing.MintingLicenseSpec calldata mintingLicenseSpec
+    ) external onlyLicensingModule {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        $.mintingLicenseSpecsForAll[ipId] = Licensing.MintingLicenseSpec({
+            isSet: true,
+            mintingFee: mintingLicenseSpec.mintingFee,
+            mintingFeeModule: mintingLicenseSpec.mintingFeeModule,
+            receiverCheckModule: mintingLicenseSpec.receiverCheckModule,
+            receiverCheckData: mintingLicenseSpec.receiverCheckData
+        });
+        emit MintingLicenseSpecSetForAll(ipId, mintingLicenseSpec);
+    }
+
+    function attachLicenseConfigToIp(
+        address ipId,
+        address licenseTemplate,
+        uint256 licenseConfigId
+    ) external onlyLicensingModule {
+        if (!_existsLicenseConfig(licenseTemplate, licenseConfigId)) {
+            revert Errors.LicensingModule__LicenseConfigNotFound(licenseTemplate, licenseConfigId);
+        }
+
+        if (_isDerivativeIp(ipId)) {
+            revert Errors.LicensingModule__DerivativesCannotAddLicenseConfig();
+        }
+
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if ($.expireTimes[ipId] < block.timestamp) {
+            revert Errors.LicenseRegistry__IpExpired(ipId);
+        }
+        $.attachedLicenseConfigs[ipId].add(licenseConfigId);
+    }
+
+    // solhint-disable-next-line code-complexity
+    function registerDerivativeIp(
+        address derivativeIpId,
+        address[] calldata originalIpIds,
+        address licenseTemplate,
+        uint256[] calldata licenseConfigIds
+    ) external onlyLicensingModule {
+        if (originalIpIds.length == 0) {
+            revert Errors.LicenseRegistry__NoOriginalIp();
+        }
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if ($.attachedLicenseConfigs[derivativeIpId].length() > 0) {
+            revert Errors.LicenseRegistry__DerivativeIpAlreadyHasLicense(derivativeIpId);
+        }
+        if ($.originalIps[derivativeIpId].length() > 0) {
+            revert Errors.LicenseRegistry__DerivativeAlreadyRegistered(derivativeIpId);
+        }
+
+        for (uint256 i = 0; i < originalIpIds.length; i++) {
+            if ($.disputeModule.isIpTagged(originalIpIds[i])) {
+                revert Errors.LicenseRegistry__OriginalIpTagged(originalIpIds[i]);
+            }
+            if (derivativeIpId == originalIpIds[i]) {
+                revert Errors.LicenseRegistry__DerivativeIsOriginal(derivativeIpId);
+            }
+            if ($.expireTimes[originalIpIds[i]] < block.timestamp) {
+                revert Errors.LicenseRegistry__OriginalIpExpired(originalIpIds[i]);
+            }
+            // derivativeIp can only register with default license terms or the same license terms as the original IP
```

Comment:
Could you explain this comment? Could the default license terms the be incompatible with the original Ip terms?

Reply:
As the default license terms is social remix terms. afaik, we don't consider the default license terms could incompatible with parent IP terms. We might upgrade this rule if we consider otherwise later.